bindings/c/test/unit/unit_tests.cpp

```diff
@@ -0,0 +1,1840 @@
+/*
+ * unit_tests.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unit tests for the FoundationDB C API.
+
+#define FDB_API_VERSION 620
+#include <foundationdb/fdb_c.h>
+#include <assert.h>
+#include <string.h>
+
+#include <condition_variable>
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <optional>
+#include <stdexcept>
+#include <thread>
+#include <tuple>
+#include <vector>
+
+#define DOCTEST_CONFIG_IMPLEMENT
+#include "doctest.h"
+
+#include "fdb_api.hpp"
+
+// Macros to replace bare key with prefixed key.
+#define KEY(key) (prefix + key)
```

Comment:
These don't need to be macros, they can be functions. Even then we probably want to convert api's to use std::string_view so we don't need to duplicate the key at every call site